src/api.rs

```diff
@@ -1485,4 +1495,316 @@ mod test {
 
     assert_eq!(limit, count);
   }
+
+  fn encode_frames<T: Pixel>(
+    ctx: &mut Context<T>, limit: u64, scene_change_at: u64
+  ) -> Vec<(u64, bool)> {
+    for i in 0..limit {
+      let mut input = Arc::try_unwrap(ctx.new_frame()).unwrap();
+
+      if i < scene_change_at {
+        fill_frame_const(&mut input, T::min_value());
+      } else {
+        fill_frame_const(&mut input, T::max_value());
+      }
+
+      let _ = ctx.send_frame(Arc::new(input));
+    }
+
+    ctx.flush();
+
+    (0..)
```

Comment:
`ctx.inner.frame_invariants.values()` maybe does the same in a simpler way.

Reply:
Eh, I'm not sure. Since here I'm calling `set_frame_properties` every step (which actually fills `frame_invariants` one by one) and also collecting the `end_of_subgop` values it returns (which are not saved anywhere else) in addition to the frame invariants themselves. So to use `frame_invariants.values()` I'd have to collect all `end_of_subgop` values separately using basically the same code, and then additionally zip them with `frame_invariants.values()` or something.

Reply:
We can play code golf on it later :)